File: H5/H5/Resources/.generated/System/ComponentModel/BrowsableAttribute.js
    H5.define("System.ComponentModel.BrowsableAttribute", {
        inherits: [System.Attribute],
        statics: {
            fields: {
                yes: null,
                no: null,
                default: null
            },
            ctors: {
                init: function () {
                    this.yes = new System.ComponentModel.BrowsableAttribute(true);
                    this.no = new System.ComponentModel.BrowsableAttribute(false);
                    this.default = System.ComponentModel.BrowsableAttribute.yes;
                }
            }
        },
        fields: {
            browsable: false
        },
        props: {
            Browsable: {
                get: function () {
                    return this.browsable;
                }
            }
        },
        ctors: {
            init: function () {
                this.browsable = true;
            },
            ctor: function (browsable) {
                this.$initialize();
                System.Attribute.ctor.call(this);
                this.browsable = browsable;
            }
        },
        methods: {
            equals: function (obj) {
                if (H5.referenceEquals(obj, this)) {
                    return true;
                }
                var other;

                return (((other = H5.as(obj, System.ComponentModel.BrowsableAttribute))) != null) && other.Browsable === this.browsable;
            },
            getHashCode: function () {
                return H5.getHashCode(this.browsable);
            }
        }
    });

File: H5/H5/System/ComponentModel/BrowsableAttribute.cs
namespace System.ComponentModel
{
    using System;

    /// <summary>
    /// Specifies whether a property or event should be displayed in a Properties window.
    /// </summary>
    [H5.Convention(Member = H5.ConventionMember.Field | H5.ConventionMember.Method, Notation = H5.Notation.CamelCase)]
    [AttributeUsage(AttributeTargets.All)]
    public sealed class BrowsableAttribute : Attribute
    {
        /// <summary>
        /// Specifies that a property or event can be modified at design time. This field is read-only.
        /// </summary>
        public static readonly BrowsableAttribute Yes = new BrowsableAttribute(true);

        /// <summary>
        /// Specifies that a property or event cannot be modified at design time. This field is read-only.
        /// </summary>
        public static readonly BrowsableAttribute No = new BrowsableAttribute(false);

        /// <summary>
        /// Specifies the default value for the BrowsableAttribute, which is BrowsableAttribute.Yes. This field is read-only.
        /// </summary>
        public static readonly BrowsableAttribute Default = Yes;

        private bool browsable = true;

        /// <summary>
        /// Initializes a new instance of the BrowsableAttribute class.
        /// </summary>
        /// <param name="browsable"></param>
        public BrowsableAttribute(bool browsable)
        {
            this.browsable = browsable;
        }

        /// <summary>
        /// Gets a value indicating whether an object is browsable.
        /// </summary>
        public bool Browsable
        {
            get
            {
                return browsable;
            }
        }

        public override bool Equals(object obj)
        {
            if (obj == this)
            {
                return true;
            }


            return (obj is BrowsableAttribute other) && other.Browsable == browsable;
        }

        public override int GetHashCode()
        {
            return browsable.GetHashCode();
        }
    }
}